pkg/apis/submariner.io/v1/string.go

```diff
@@ -0,0 +1,17 @@
+package v1
+
+import (
+	"encoding/json"
+
+	"k8s.io/klog"
+)
+
+// String method returns a string representation, for human readability.
+func (e *Endpoint) String() string {
+
+	bytes, err := json.Marshal(e.Spec)
+	if err != nil {
+		klog.Fatalf("Error serializing Endpoint spec to string: %s", err.Error())
```

Comment:
I think it's rather drastic to exit the process for a debug string error. If it fails then format it with "%#v".